pkg/endpointmanager/manager.go

```diff
@@ -405,12 +405,20 @@ func AddEndpoint(owner endpoint.Owner, ep *endpoint.Endpoint, reason string) (er
 	}
 
 	if build {
-		go func() {
+		if asynchronouslyBuild {
```

Comment:
I don't see a good reason to add the logic of building the endpoint async or not into the endpoint manager. Besides, the endpoint itself offers a function to perform the regeneration async, `ep.Regenerate`.

Reply:
I did this to maintain existing behavior on the Docker plugin side, so I would be reducing the amount of changes introduced in this PR. I also kept the `RegenerateWait` call in so we would remove the endpoint in the async case if the first build failed, as was being done before since I didn't want to change existing behavior too much in that case as well.